ibm/service/cis/resource_ibm_cis_origin_auth.go

```diff
@@ -0,0 +1,301 @@
+)
+
+func ResourceIBMCISOriginAuthPull() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCISOriginAuthPullCreate,
+		ReadContext:   resourceIBMCISOriginAuthPullRead,
+		UpdateContext: resourceIBMCISOriginAuthPullUpdate,
+		DeleteContext: resourceIBMCISOriginAuthPullDelete,
+		Importer:      &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Optional:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisOriginAuthHost: {
+				Type:        schema.TypeString,
+				Description: "Host name needed for host level authentication",
+				Optional:    true,
+			},
+			cisOriginAuthEnable: {
+				Type:        schema.TypeBool,
+				Description: "Enabel-disable origin auth for a zone or host",
+				Optional:    true,
+				Default:     true,
+			},
+			cisOriginAuthCertContent: {
+				Type:        schema.TypeString,
+				Description: "Certificate content which needs to be uploaded",
+				Required:    true,
+			},
+			cisOriginAuthCertKey: {
+				Type:        schema.TypeString,
+				Description: "Private key content which needs to be uploaded",
+				Required:    true,
+			},
+			CisOriginAuthStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Authentication status whether active or not",
+			},
+			cisOriginAuthCertId: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate ID which is uploaded",
+			},
+			cisOriginAuthExpiresOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate expires on",
+			},
+			cisOriginAuthUploadedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate uploaded on",
+			},
+			cisOriginAuthID: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS auth pull job id",
+				Computed:    true,
+			},
+		},
+	}
+
+}
+
+func resourceIBMCISOriginAuthPullCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var cert_val string
+	var key_val string
+	var zone_config bool
+
+	sess, err := meta.(conns.ClientSession).CisOrigAuthSession()
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error while getting the CisOrigAuthSession %v", err))
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, _ := flex.ConvertTftoCisTwoVar(d.Get(cisDomainID).(string))
+
+	sess.ZoneIdentifier = core.StringPtr(zoneID)
+	sess.Crn = core.StringPtr(crn)
+
+	if cert_content, ok := d.GetOk(cisOriginAuthCertContent); ok {
+		cert_val = cert_content.(string)
+
+	}
+
+	if cert_key, ok := d.GetOk(cisOriginAuthCertKey); ok {
+		key_val = cert_key.(string)
+
+	}
+	zone_config = true
+	if _, ok := d.GetOk(cisOriginAuthHost); ok {
+		zone_config = false
```

Comment:
Instead of calculating zone_config this way can't we use level with default "zone" (simialr to cli https://cloud.ibm.com/docs/cis?topic=cis-cli-plugin-cis-cli#upload-authenticated-origin-pull-certificate)
and while settign back the ID add this info also 

d.SetId(flex.ConvertCisToTfThreeVar(*result.Result.ID, level, zoneID, crn)) and you should remove d.GetOk(cisOriginAuthHost) in Read, update, delete

Reply:
OK. I'll check it further.